### ISSUE TITLE
Authenticate to restnotification API using OAuth2

### ISSUE THREAD

**Issue by yanniks** [SAPDocuments/Tutorials#2754]
Hello,

We're currently trying to authenticate to the restnotification API using OAuth2. I followed [this](https://blogs.sap.com/2017/06/09/how-to-enable-push-notifications-（part2/) tutorial, unfortunately it doesn't say how to authenticate using OAuth2. We're using our own SAML-IdP in connection with the OAuth2 service. I created a new client for our "push notification handling website", obtaining access tokens is working fine.
Unfortunately, I cannot use that token to authenticate against e. g. `https://mobile-ID.hana.ondemand.com/restnotification/application/APPID/user/USERNAME`, I always get a 401 response. Basic authentication is rarely an option since we'd like to offer authentication using our internal credentials instead of S-users.

I already found issue #2404, unfortunately it was never solved.

Is there a solution to that problem?

Thanks in advance,
Yannik

**Comment by qualiture** [comment]
Hi @KevinMuessig, 

As you are probably closer to the development team, could you maybe forward this to them and see if they know a solution?

Thanks in advance!

**Comment by KevinMuessig** [comment]
Hi @yanniks,

Apologizes for taking care of that so late.

Please let me know if this is still an issue.

Thanks,

Kevin

**Comment by yanniks** [comment]
Hi @KevinMuessig,

no problem.
Unfortunately, this problem still exists, but it was not a high priority for us as we are not yet sending push notifications. For testing, I authenticated using Basic Authentication but OAuth would be the preferable way to go. When creating the OAuth Client, I selected `hanamobileprod/mobilejava` as subscription type, maybe that's wrong?

Thanks,
Yannik

**Comment by KevinMuessig** [comment]
Hi @yanniks ,

I have written a tutorial for push notifications with the SAPcp iOS SDK and Mobile Services. Could you take a look at that, it may help.

Are you writing an iOS app with the SDK or what is connecting against mobile services?

https://developers.sap.com/tutorials/fiori-ios-hcpms-push-notifications.html

Thanks

**Comment by thecodester** [comment]
Thank you for your feedback. We haven't heard from you in a while, so we are closing the issue. 

If you still have questions, feel free to reopen the issue.

**Comment by yanniks** [comment]
Hi @KevinMuessig , thanks for your feedback. The question was not really targeted on how to get the notification from the SCP to the user's device but instead on how to submit new messages to the SCP - that might happen from anything that can issue a HTTP POST request such as a Java application, a website or something else.

Mobile Services provides a REST web service which can be used in order to submit new push notifications to single users, all users of an application or a specific group. Unfortunately, I only figured out how to authenticate using basic authentication. Instead of that, I would like to use OAuth2 - one could configure a new client in the SMP and the bearer token retrieved from the token endpoint could be used in order to authenticate at that REST endpoint. Unfortunately, that didn't work for me when I tried it.

Is this supposed to work? This would provide a big benefit for users and security, especially if one uses a custom Identity Provider for authentication instead of the default SAP S-Users.
